packages/react-core/src/components/ContextSelector/examples/ContextSelector.md

```diff
@@ -82,3 +82,91 @@ class SimpleContextSelector extends React.Component {
   }
 }
 ```
+
+
+### With component before search bar
+```js 
+import React from 'react';
+import { ContextSelector, ContextSelectorItem, Title } from '@patternfly/react-core';
+
+class ContextSelectorWithBeforeSearch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.items = [
+      'My Project',
+      'OpenShift Cluster',
+      'Production Ansible',
+      'AWS',
+      'Azure',
+      'My Project 2',
+      'OpenShift Cluster ',
+      'Production Ansible 2 ',
+      'AWS 2',
+      'Azure 2'
+    ];
+
+    this.state = {
+      isOpen: false,
+      selected: this.items[0],
+      searchValue: '',
+      filteredItems: this.items
+    };
+
+    this.onToggle = (event, isOpen) => {
+      this.setState({
+        isOpen
+      });
+    };
+
+    this.onSelect = (event, value) => {
+      this.setState({
+        selected: value,
+        isOpen: !this.state.isOpen
+      });
+    };
+
+    this.onSearchInputChange = value => {
+      this.setState({ searchValue: value });
+    };
+
+    this.onSearchButtonClick = event => {
+      const filtered =
+        this.state.searchValue === ''
+          ? this.items
+          : this.items.filter(str => str.toLowerCase().indexOf(this.state.searchValue.toLowerCase()) !== -1);
+
+      this.setState({ filteredItems: filtered || [] });
+    };
+  }
+
+  render() {
+    const { isOpen, selected, searchValue, filteredItems } = this.state;
+    return (
+      <ContextSelector
+        toggleText={selected}
+        onSearchInputChange={this.onSearchInputChange}
+        isOpen={isOpen}
+        searchInputValue={searchValue}
+        onToggle={this.onToggle}
+        onSelect={this.onSelect}
+        onSearchButtonClick={this.onSearchButtonClick}
+        screenReaderLabel="Selected Project:"
+        beforeSearch={
```

Comment:
can you rename it to something like `menuHeader`?

Reply:
Yeah, thanks for the suggestion